tests/admin_inlines/tests.py

```diff
@@ -86,6 +87,23 @@ def test_inline_primary(self):
         self.assertEqual(response.status_code, 302)
         self.assertEqual(len(Fashionista.objects.filter(person__firstname='Imelda')), 1)
 
+    def test_inline_primary_with_uuids(self):
```

Comment:
You mentioned this test failed during one of your trial fixes. Did none of the `model_formsets` tests fail in that case? If so, I wonder if this covers a different case that could be encapsulated in a `model_formset` test. It would be nice to avoid the additional complexity of these tests, particularly since the fix doesn't touch any admin specific code.